core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -1407,18 +1424,21 @@ private void buildMatViewGraph() {
                         }
 
                         final MatViewRefreshState state = matViewGraph.addView(matViewDefinition);
-                        final boolean isMatViewStateExists = TableUtils.isMatViewStateFileExists(configuration, path, tableToken.getDirName());
-                        path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
-                        if (isMatViewStateExists) {
-                            reader.of(path.$());
-                            MatViewRefreshState.readFrom(reader, state);
-                        } else {
-                            blockFileWriter.of(path.$());
-                            MatViewRefreshState.append(state, blockFileWriter);
-                        }
+                        // can be null if the graph has no-op implementation
+                        if (state != null) {
+                            final boolean isMatViewStateExists = TableUtils.isMatViewStateFileExists(configuration, path, tableToken.getDirName());
+                            path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
+                            if (isMatViewStateExists) {
```

Comment:
also, for troubleshooting purposes, why don't we print out names of materialized views we loaded?

Reply:
I guess that's to avoid log pollution in scenarios when there is a lot of mat views. @eugenels WDYT?